### PR TITLE
Upstream patch for McEliece __m256i_u issue

### LINKED ISSUE #940: Current master - McEliece AVX2 is broken
MacOS 10.15.7, Xcode-12.4, current master.
```
-- The C compiler identification is AppleClang 12.0.0.12000032
-- The ASM compiler identification is Clang
-- Found assembler: /opt/local/bin/clang
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/local/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenSSL: /opt/local/lib/libcrypto.dylib (found suitable version "1.1.1j", minimum required is "1.1.1")  
-- Looking for OPENSSL_NO_EC2M
-- Looking for OPENSSL_NO_EC2M - not found
-- Found Doxygen: /opt/local/bin/doxygen (found version "1.9.1") found components: doxygen dot 
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/ur20980/src/liboqs/build
[1272/2236] Building ASM object src/kem/classic_mceliece/CMakeFile...ic_mceliece_8192128_avx.dir/pqclean_mceliece8192128_avx/consts.S.
FAILED: src/kem/classic_mceliece/CMakeFiles/classic_mceliece_8192128_avx.dir/pqclean_mceliece8192128_avx/consts.S.o 
/opt/local/bin/clang  -Iinclude -I../src/kem/classic_mceliece/pqclean_mceliece8192128_avx -I../src/common/pqclean_shims -O3 -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX11.1.sdk -mmacosx-version-min=10.15 -fPIC -Werror -Wall -Wextra -Wpedantic -Wno-unused-command-line-argument -O3 -fomit-frame-pointer -mavx2 -mpopcnt -Wno-language-extension-token -MD -MT src/kem/classic_mceliece/CMakeFiles/classic_mceliece_8192128_avx.dir/pqclean_mceliece8192128_avx/consts.S.o -MF src/kem/classic_mceliece/CMakeFiles/classic_mceliece_8192128_avx.dir/pqclean_mceliece8192128_avx/consts.S.o.d -o src/kem/classic_mceliece/CMakeFiles/classic_mceliece_8192128_avx.dir/pqclean_mceliece8192128_avx/consts.S.o -c ../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:19:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK0_0
^
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:20:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK0_1
^
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:21:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK1_0
^
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:22:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK1_1
^
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:23:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK2_0
^
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:24:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK2_1
^
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:25:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK3_0
^
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:26:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK3_1
^
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:27:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK4_0
^
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:28:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK4_1
^
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:29:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK5_0
^
../src/kem/classic_mceliece/pqclean_mceliece8192128_avx/consts.S:30:1: error: unknown directive
.hidden PQCLEAN_MCELIECE8192128_AVX_MASK5_1
^
[1293/2236] Building C object src/kem/sike/CMakeFiles/sike.dir/external/P610/P610_compressed.c.o
ninja: build stopped: subcommand failed.
```

How do I configure the build to **exclude** algorithms?

### PR DESCRIPTION
**Pull request by jschanck**
This reverts the local changes to McEliece, adds a upstream patch to fix the __m256i_u issue, and pulls the latest version of McEliece using copy_from_upstream. Pulling from PQClean resolves #940. We still need to do another pull from PQClean to get the latest version of SPHINCS.

### CODE REVIEW

File: src/kem/classic_mceliece/pqclean_mceliece348864f_avx/consts.S [hunk 1]
```diff
@@ -1,7 +1,10 @@
-.data
+#if defined(__APPLE__)
```
Comment (baentsch) [inline src/kem/classic_mceliece/pqclean_mceliece348864f_avx/consts.S:1]: Am I blind not finding this in the patch code? Or is it intentionally not part of the patch?

Comment (jschanck) [reply src/kem/classic_mceliece/pqclean_mceliece348864f_avx/consts.S:1]: That's a change from upstream pqclean.

**Comment by jschanck** [comment]
Weird that alpine failed with insufficient storage on the last run.

Adding a commit to bump the commit number for PQClean in copy_from_upstream.yml. With this commit, if you want to update SPHINCS you just have to run copy_from_upstream.

**Review by baentsch — APPROVED** [review]
